app/src/main/java/com/lms/mpasho_lms_news/MainActivity.java

```diff
@@ -1,13 +1,125 @@
 package com.lms.mpasho_lms_news;
 
-import android.support.v7.app.AppCompatActivity;
+import android.app.SearchManager;
+import android.content.Context;
 import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.DefaultItemAnimator;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.SearchView;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.widget.Toast;
+
+import com.lms.mpasho_lms_news.api.ApiClient;
+import com.lms.mpasho_lms_news.api.ApiInterface;
+import com.lms.mpasho_lms_news.models.Article;
+import com.lms.mpasho_lms_news.models.News;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
 
 public class MainActivity extends AppCompatActivity {
 
+    public static final String API_KEY = "ccf1c18922cb4be88aeb1b6a08b13a5d";
+    public RecyclerView recyclerView;
+    public RecyclerView.LayoutManager layoutManager;
+    private List<Article> articles = new ArrayList<>();
+    private Adapter adapter;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        recyclerView = findViewById(R.id.recyclerView);
+        layoutManager = new LinearLayoutManager(MainActivity.this);
+        recyclerView.setLayoutManager(layoutManager);
+        recyclerView.setItemAnimator(new DefaultItemAnimator());
+        recyclerView.setNestedScrollingEnabled(false);
+
+        LoadJson("");
+
+    }
+
+
+    public void LoadJson(final String keyword) {
+
+        ApiInterface apiInterface = ApiClient.getApiClient().create(ApiInterface.class);
+
+        String country = Utils.getCountry();
+        String language = Utils.getLanguage();
+
+        Call<News> call;
+
+        if (keyword.length() > 0) {
+            call = apiInterface.getNewsSearch(keyword, language, "publishedAt", API_KEY);
+        } else {
+            call = apiInterface.getNews(country, API_KEY);
+        }
+
+        call.enqueue(new Callback<News>() {
```

Comment:
What kind of architechture are you using? I am familiar with MVP and it would clean this up by having it in the Presenter

Reply:
I am following the MVP architecture, I will implement the `Presenter` and Model for implementing offline functionality.

Reply:
Am following the `MVP` Architecture, I'll implement the `PBF` structure for the same